wire-ios-sync-engine/Source/SessionManager/SessionManager+AnalyticsUseCases.swift

```diff
@@ -16,26 +16,39 @@
 // along with this program. If not, see http://www.gnu.org/licenses/.
 //
 
+import WireAnalytics
+
 extension SessionManager {
 
     enum AnalyticsSessionError: Error {
 
         case analyticsNotAvailable
+        case analyticsConfigurationNotAvailable
         case missingAnalyticsUserProfile
         case missingActiveUserSession
 
     }
 
     public func makeDisableAnalyticsUseCase() throws -> DisableAnalyticsUseCaseProtocol {
-        guard let analyticsManager else {
-            throw AnalyticsSessionError.analyticsNotAvailable
+
+        guard let userSession = self.activeUserSession else {
+            throw AnalyticsSessionError.missingActiveUserSession
+        }
+
+        let didDisableAnalyticsClosure: () -> Void = {
+            userSession.analyticsSession = nil
         }
-        return DisableAnalyticsUseCase(analyticsManager: analyticsManager)
+
+        return DisableAnalyticsUseCase(
```

Comment:
We are passing 2 huge objects to `DisableAnalyticsUseCase`: `sessionManager` and `userSession`? Is it so necessary? Especially that `sessionManager` already has a reference to the `activeUserSession`.

Reply:
Same question about `EnableAnalyticsUseCase`.  @netbe  and @caldrian  I would like to ask you opinions as well

Reply:
> We are passing 2 huge objects to DisableAnalyticsUseCase: sessionManager and userSession?

I don't get the point about "huge objects".
The use cases depend on protocols, one which declares an `analyticsManager` property, where the `AnalyticsManager` reference is set or cleared, and another protocol where the `analyticsSession` can be stored.
It doesn't depend on anything from `ZMUserSession` or `SessionManager`.

If/why the two instances need to be stored separately as well as any implementation detail around `WireAnalytics` I haven't looked into yet.

> Especially that sessionManager already has a reference to the activeUserSession.

Is there always one user session at a time which is available through `SessionManager. activeUserSession`? Then it might make sense to combine the protocols.

Reply:
> I can guess that you wanted to pass didDisableAnalyticsClosure instead of sessionManager, no?
yes that was a way I suggested to not mock userSession in SyncEngine